### PR TITLE
Upgrade agent to .NET 6

### PR DESCRIPTION
**Pull request by DenisRumyantsev**
- Bumped dotnet version
- Handled code analysis errors
- Fixed pipeline on Window x86

### CODE REVIEW

**Comment by anatolybolshakov** [comment]
Please don't merge until testing is completed

File: .vsts.ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ parameters:
 - name: rhel6_x64
   type: boolean
   displayName: RHEL 6 (x64)
```
Comment (anatolybolshakov) [inline .vsts.ci.yml:24]: Should this be updated to RHEL 7?

**Review by anatolybolshakov — APPROVED** [review]
LGTM in general, thanks! Please take a look at comments (and we still need to test it of course).

**Comment by nekrich** [comment]
@anatolybolshakov, @AndreyIvanov42 is there any news on this PR?
I assume it will help with [#3184 [Question] Azure Pipeline Agent support for Apple M1](https://github.com/microsoft/azure-pipelines-agent/issues/3184).

**Comment by kdubau** [comment]
@DenisRumyantsev @anatolybolshakov what is the deal here? Can we get ~any update? When is this expected to ship?

File: .vsts.ci.yml (hunk 1 above)
Comment (AndreyIvanov42) [reply .vsts.ci.yml:24]: Yes, we have other PR for that - https://github.com/microsoft/azure-pipelines-agent/pull/3771

**Comment by spyder007** [comment]
@DenisRumyantsev This PR is marked as closed but doesn't look like it has been merged?  Is there a release for this upgrade?
Running the agent on .net 3.1 with old SSL libraries is causing some issues, specifically #3834

**Pull request closed without merging**